tests/test_operations.py

```diff
@@ -1509,5 +1511,50 @@ def test_invalid_secrets(self):
             s.run()
 
 
+@testmode.integration
```

Comment:
Nicely written tests, the API design makes this easy to follow.

Reply:
Note - one thing you could improve on in these tests is testing passing INVALID inputs, to ensure they fail as expected, or to try passing single AND multiple inputs, to ensure that both work as expected.

And of course more comments & docstrings.